### PR TITLE
Update the tests to use the new fusion and padding transform ops.

### PR DESCRIPTION
**Pull request by gysit**

### CODE REVIEW

File: python/examples/core/harness.py [hunk 1]
```diff
@@ -592,7 +592,7 @@ def test_harness(problem_factory: Callable[
 
   measurements = Measurements()
 
-  backends = kwargs.get("backends", ['strategy'])
+  backends = kwargs.get("backends", ['dialect'])
```
Comment (nicolasvasilache) [inline python/examples/core/harness.py:595]: this should now be gone at main

File: python/examples/core/transforms.py [hunk 2]
```diff
@@ -20,9 +20,9 @@ def _get_size_list_as_str(name: str, sizes: tp.List[int]) -> str:
 
 def _get_pad_str(transform: Transform) -> str:
   """Compute the textual padding flags for the given `transform`."""
-  if not transform.pad:
+  if hasattr(transform, 'pad') and not transform.pad:
```
Comment (nicolasvasilache) [inline python/examples/core/transforms.py:23]: this should now be gone at main

File: python/examples/core/transforms.py [hunk 3]
```diff
@@ -104,55 +104,36 @@ class Fuse(Transform):
   This transform can be configured as follows:
   * `tile_sizes`: Tile sizes used for tiling.
   * `tile_interchange`: Interchange used for tiling.
-  * `pad`: Pad the operands.
```
Comment (nicolasvasilache) [inline python/examples/core/transforms.py:107]: nice!

File: python/examples/core/transforms.py [hunk 4]
```diff
@@ -240,6 +221,49 @@ def build_transform_ir(self):
...
+      'hoist_paddings': (HoistPaddingVariable, []),
+      'transpose_paddings': (TransposePaddingVariable, []),
+  }
+
+  def __init__(self, fun_name: str, op_name: str, **kwargs):
+    self._parse_variables_in_kwargs(kwargs)
+    self.fun_name = fun_name
+    self.op_name = op_name
+    pad_str = _get_pad_str(self)
+    pipeline = (f'linalg-single-tiling-expert-driver{{'
```
Comment (nicolasvasilache) [inline python/examples/core/transforms.py:250]: this should now be gone at main

File: python/examples/fusion/test.py [hunk 5]
```diff
@@ -17,11 +17,11 @@
 def fill_matmul_fusion():
   fun_name = 'matmul'
   op_name = 'linalg.matmul'
-  # FIXME: FusionOp does not implement build_transform_ir
-  # Fuse(fun_name, op_name, tile_sizes=[8, 16, 0], tile_interchange=[])
-  expert = Tile(fun_name, 'linalg.fill', pad=True,)                \
-    .then(Tile(fun_name, op_name, tile_sizes=[0, 0, 24], pad=True, \
-               pack_paddings=[1, 1, 0],))                          \
+  expert = Fuse(fun_name, op_name, tile_sizes=[8, 16, 0],          \
```
Comment (nicolasvasilache) [inline python/examples/fusion/test.py:20]: nice!

**Review by nicolasvasilache — APPROVED** [review]
Will do a few more cleanups at main by the time you land this but you can ignore anything related to string stitching of pass pipelines.

**Review by nicolasvasilache — APPROVED** [review]
Update the FIXME too please ?

**Comment by gysit** [comment]
Done!

**Pull request merged**